src/Decimal.php

```diff
@@ -27,7 +26,7 @@ class Decimal implements JsonSerializable
      *
      * Value before the separator. Cannot be negative.
      *
-     * @var int
+     * @var string
```

Comment:
I thought it would be much easier to support big decimals and integers by using `string` for the integer part.
Pros:
- No casting needed from integer to float or vice-versa, we only cast anything to string, which is most a safe cast.
- We didn't have any check for big integer inputs, in scientific and non-scientific format, casting string inputs to integer or float results in different numbers (mostly max integer, or min integer).
- Scientific values can be a backdoor to support big integer, if not checked and compared against the integer boundaries.

Cons:
- `toInt` and `toFloat` have to throw an exception for big integers and decimals.

Reply:
For example, null was allowed because it gets casted to string first, then to an integer.
`(string)null => ''` and then to an integer `(int)'' => 0`, which is a funny behaviour.